scripts/gen-zap-script/utils/query.py

```diff
@@ -0,0 +1,83 @@
+class Query:
+    @staticmethod
+    def _build_query(data):
+        """Build query object from data.
+
+        data: may be any python structure
+
+        Return:
+            an instance of `Item` class
+        """
+
+        def walk_data(data):
+            children = []
+            if isinstance(data, dict):
+                for key, value in data.items():
+                    children.append(Item(name=key, children=walk_data(value)))
+            elif isinstance(data, list):
+                for value in data:
+                    if isinstance(value, (dict, list)):
+                        children.append(Item(children=walk_data(value)))
+                    else:
+                        children.append(Item(value=value))
+            else:
+                children.append(Item(value=data))
+            return children
+
+        return Item(children=walk_data(data))
+
+
+class Item:
+    """A class which represents a node of the query object."""
+
+    def __init__(self, name=None, children=None, value=None):
+        self._name = name
+        self._children = children
+        self._value = value
+
+    def __getattr__(self, attr_name):
+        if self.children:
+            for child in self.children:
+                if isinstance(child, Item):
+                    if child.name == attr_name:
+                        return child
+                if isinstance(child, list):
+                    return child
+
+        return Item()
+
+    def __getitem__(self, index):
+        if isinstance(index, int) and self.children and len(self.children) >= index:
+            return self.children[index]
+        if isinstance(index, str):
+            return getattr(self, index)
+        raise ValueError("The index value should be int or string")
+
+    @property
+    def children(self):
+        return self._children
+
+    @property
+    def name(self):
+        return self._name
+
+    @property
+    def value(self):
+        if self.children and isinstance(self.children, list):
+            value_list = [i.value for i in self.children]
+            if len(value_list) == 1:
+                return value_list[0]
```

Comment:
I haven't tested this yet but if this will be applied globally, I think it will break current functionality by returning a string when a list is expected. For example https://github.com/RedHatProductSecurity/rapidast/blob/development/scripts/apis_scan.py#L64 will iterate over the string chars instead of the list values